### PR TITLE
Update CRT submodules and document the process

### PR DESCRIPTION
**Pull request by passaro**
Address issue #219.

Update CRT submodules to the latest tagged releases:

* aws-c-auth v0.6.26
* aws-c-cal v0.5.26
* aws-c-common v0.8.17
* aws-c-compression v0.2.16
* aws-c-http v0.7.7
* aws-c-io v0.13.21
* aws-c-s3 v0.2.8
* aws-c-sdkutils v0.1.9
* aws-checksums v0.1.14
* aws-lc v1.9.0
* s2n-tls v1.3.42

Also add short document to describe the process.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: mountpoint-s3-crt-sys/UPDATING_CRT.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+* `aws-c-cal`
+* `aws-c-common`
+* `aws-c-compression`
+* `aws-c-http`
+* `aws-c-io`
+* `aws-c-s3`
+* `aws-c-sdkutils`
+* `aws-checksums`
+* `aws-lc`
+* `s2n-tls`
```
Comment (dannycjones) [inline mountpoint-s3-crt-sys/UPDATING_CRT.md:17]: I'd drop this, let the `/crt` directory document itself.

File: mountpoint-s3-crt-sys/UPDATING_CRT.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Update Submodules
+
+The CRT submodules can be updated by following these steps:
+
+1. Update every submodule to the latest tagged release. E.g. by running:
+
+   ```sh
+   git submodule foreach -q 'git checkout `git tag -l --sort=-v:refname | head -1`'
+   ```
```
Comment (dannycjones) [inline mountpoint-s3-crt-sys/UPDATING_CRT.md:27]: I think we need to add in a fetch here. Also, let's not suppress the `submodule foreach` log telling us which submodule the update is applied to - I think it'll help make the output clearer.

```suggestion
1. Update every submodule to the latest tagged release. E.g. by running:

   ```sh
   git submodule foreach 'git fetch -q --tags && git checkout `git tag -l --sort=-v:refname | head -1`'
   ```
```

**Review by dannycjones — COMMENTED** [review]
Looks good, just some small suggestions

File: mountpoint-s3-crt-sys/UPDATING_CRT.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+2. Review the commit history, in particular for `aws-c-s3`, looking for changes that may affect `mountpoint-s3` (bug fixes, API changes, etc.).
```
Comment (dannycjones) [inline mountpoint-s3-crt-sys/UPDATING_CRT.md:29]: Should we recommend reviewing the output of `git diff --submodule [path to submodule]`?

For example:

```
❯ git diff --submodule mountpoint-s3-crt-sys/crt/aws-c-s3 

Submodule mountpoint-s3-crt-sys/crt/aws-c-s3 d7bfe60..ae6c6cc:
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
  > Fix one byte range handling (#272)
  > Fix Benchmark stack (#270)
  > Update package.json (#260)
  > New `send_filepath` option, for faster uploads. (#266)
  > Skip mock server setup when tests are disabled (#267)
...
```

File: mountpoint-s3-crt-sys/UPDATING_CRT.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+* `aws-c-io`
+* `aws-c-s3`
+* `aws-c-sdkutils`
+* `aws-checksums`
+* `aws-lc`
+* `s2n-tls`
+
+## Update Submodules
+
+The CRT submodules can be updated by following these steps:
```
Comment (dannycjones) [inline mountpoint-s3-crt-sys/UPDATING_CRT.md:21]: Step 4: commit the updated changes? Is it a case of add and commit?

**Comment by dannycjones** [comment]
I also reviewed the commits from the CRT we are pulling in. They look OK, but it is difficult to feel confident about the review.

I wonder if we want to automate pulling in new CRT libs on some cadence, as some of these packages have _a lot of commits_ to pull in.

File: mountpoint-s3-crt-sys/UPDATING_CRT.md (hunk 2 above)
Comment (passaro) [reply mountpoint-s3-crt-sys/UPDATING_CRT.md:27]: You're right. I only meant to add it to the "check" command where we already print the submodule path.

File: mountpoint-s3-crt-sys/UPDATING_CRT.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+   git diff --submodule mountpoint-s3-crt-sys/crt/aws-c-s3 
+   ```
+
+3. Build and test `mountpoint-s3`.
+
+4. Stage and commit the changes:
+
+   ```sh
+   git commit -a -s -m "Update CRT submodules to latest releases"
+   ```
```
Comment (dannycjones) [inline mountpoint-s3-crt-sys/UPDATING_CRT.md:27]: Unless obvious, I'd generally recommend to use the full flag names in documentation.

For instance, something like:

   ```sh
   git commit --all --signoff \
       -m "Update CRT submodules to latest releases"
   ```

I'm also a bit wary of recommending the use of `--all` flag, but fine for this.

Not blocking this PR on any of this.

**Review by dannycjones — DISMISSED** [review]
LGTM

**Review by dannycjones — APPROVED** [review]
LGTM, thanks Alex!

**Pull request merged**